controllers/feeds/getFeedChat.js

```diff
@@ -148,12 +148,18 @@ const finalize = (req, id, myReaction, newGroup, activity_id, constantActor, kar
       (data) => data.reaction.kind === 'comment'
     );
     if (listComment.length > 0) {
-      newGroup[activity_id].data.last_message_at = listComment?.sort(
-        (a, b) => new Date(a.reaction.created_at) - new Date(b.reaction.created_at)
-      )?.[0]?.reaction?.created_at;
-      newGroup[activity_id].data.updated_at = listComment?.sort(
-        (a, b) => new Date(a.reaction.updated_at) - new Date(b.reaction.updated_at)
-      )?.[0]?.reaction?.created_at;
+      const sortedCommentByCreatedAt = listComment.sort(
+        (a, b) => new Date(b.reaction.created_at) - new Date(a.reaction.created_at)
+      );
+
+      const sortedCommentByUpdatedAt = listComment.sort(
+        (a, b) => new Date(b.reaction.updated_at) - new Date(a.reaction.updated_at)
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a performance issue in this code. You are sorting the `listComment` array twice, once by `created_at` and then by `updated_at`. This results in a time complexity of O(n log n) for each sort operation, which can be costly if the list is large. Instead, you could find the maximum `created_at` and `updated_at` in one pass using a single loop, reducing the time complexity to O(n). Here's how you can do it:

```diff
- const sortedCommentByCreatedAt = listComment.sort(
-   (a, b) => new Date(b.reaction.created_at) - new Date(a.reaction.created_at)
- );
- 
- const sortedCommentByUpdatedAt = listComment.sort(
-   (a, b) => new Date(b.reaction.updated_at) - new Date(a.reaction.updated_at)
- );

+ let maxCreatedAt = listComment[0].reaction.created_at;
+ let maxUpdatedAt = listComment[0].reaction.updated_at;
+ 
+ for (let i = 1; i < listComment.length; i++) {
+   if (new Date(listComment[i].reaction.created_at) > new Date(maxCreatedAt)) {
+     maxCreatedAt = listComment[i].reaction.created_at;
+   }
+   if (new Date(listComment[i].reaction.updated_at) > new Date(maxUpdatedAt)) {
+     maxUpdatedAt = listComment[i].reaction.updated_at;
+   }
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->